src/main/js/field/TableDisplay.js

```diff
@@ -134,9 +136,25 @@ class TableDisplay extends Component {
         this.updateData();
     }
 
+    handleInsertModalSubmit(event, onModalClose) {
+        if (event) {
+            event.preventDefault()
+            event.stopPropagation();
+        }
```

Comment:
Do we need this if statement if we're calling it in the handleSubmit as well?

Reply:
Have to stop the event propagation before the onModalClose otherwise all the modals are closed on this one event. Also handleSubmit is called by other functions.  So we have to stop the even propagation in both places.